controllers/datafoundationclient.go

```diff
@@ -52,16 +55,34 @@ func dfcSetupWatches(controllerBuilder *builder.Builder) {
 	)
 	controllerBuilder.
 		Owns(&opv1a1.ClusterServiceVersion{}, csvPredicates).
-		Owns(&ocsclient.StorageClient{}).
-		Owns(&ocsclient.StorageClassClaim{}).
-		Owns(&corev1.ConfigMap{})
+		Owns(&corev1.ConfigMap{}).
+		Owns(&opv1a1.ClusterServiceVersion{}, csvPredicates)
+
+	if offeringReconciler.AvailableCRDs[StorageClientCRD] {
+		controllerBuilder.
+			Owns(&ocsclient.StorageClient{})
+	}
+	if offeringReconciler.AvailableCRDs[StorageClassClaimCRD] {
+		controllerBuilder.
+			Owns(&ocsclient.StorageClassClaim{})
+	}
 }
 
 func DFCAddToScheme(scheme *runtime.Scheme) {
 	utilruntime.Must(ocsclient.AddToScheme(scheme))
 }
 
 func dfcReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+
+	if !offeringReconciler.AvailableCRDs[StorageClientCRD] {
+		// Check if the required CRDs are installed or not
+		crd := ocsclient.StorageClient{}
+		err := offeringReconciler.get(&crd)
+		if err != nil && !errors.IsNotFound(err) {
+			return ctrl.Result{}, &utils.ErrAgentRestart{Msg: "restart agent to register watchers for the CRDs"}
+		}
```

Comment:
The logic seems broken here. You are restarting if there is a get error that is not a NotFound error.
The correct logic is: Get the CRD, if the get operation results in any kind of error then request via error. 
If the Get operation was successful then restart